src/ucs/sys/compiler_def.h

```diff
@@ -135,7 +137,7 @@
  * @return Address of the container structure.
  */
 #define ucs_container_of(_ptr, _type, _member) \
-    ( (_type*)( (char*)(void*)(_ptr) - ucs_offsetof(_type, _member) )  )
+    ( (_type*)( (char*)(void*)((uintptr_t)(_ptr) - ucs_offsetof(_type, _member)) )  )
```

Comment:
can we remote `(char*)` part?

Reply:
I believe we can, trying to figure out the issue below. Any idea what causes it?

@azure-pipelines
UCX PR (Build Static checks) Failing after 17m — Build Static checks failed

Reply:
@yosefe , Seems like the issue below isn't related to this commit:
@azure-pipelines
UCX PR (Build Static checks) Failing after 17m — Build Static checks failed

The issue seems to be the fact that no NULL checking is performed on the qp and qp->verbs.rd pointers.

----- issue ---
usr/bin/bash --noprofile --norc /__w/_temp/b4059e5d-6761-4931-b004-8d95518c4abc.sh
static checkers found errors:
Error: CLANG_WARNING:
/src/uct/ib/mlx5/ib_mlx5.c:12: included_from: Included from here.
/src/uct/ib/mlx5/ib_mlx5.inl:532:26: warning: The left operand of '!=' is a garbage value
#    if (qp->verbs.rd->pd != NULL) {
#                         ^

Error: CLANG_WARNING:
/src/uct/ib/rc/accel/rc_mlx5_common.c:943:33: warning: The left operand of '!=' is a garbage value
#    ucs_assert(iface->dm.dm->dm != NULL);
#               ~~~~~~~~~~~~~~~~ ^

Reply:
@shuki-zanyovka did you see such error on any other PR?

Reply:
No, it's somehow related to this PR, not sure what's the reason.